raiden/raiden_service.py

```diff
@@ -288,18 +288,13 @@ def __init__(
         self.message_handler = message_handler
         self.blockchain_events: Optional[BlockchainEvents] = None
 
+        self.api_server: Optional[APIServer] = api_server
         self.raiden_api: Optional[RaidenAPI] = None
         self.rest_api: Optional[RestAPI] = None
-        self.api_server: Optional[APIServer] = None
-        if config.rest_api.rest_api_enabled:
+        if isinstance(api_server, APIServer):
```

Comment:
I would prefer
```suggestion
        if api_server is not None:
```
since this is about being set and not about it being an instance of a certain class.

Reply:
Same as above, this adds a type hint for mypy.